client/my-sites/checkout/index.js

```diff
@@ -107,6 +108,17 @@ export default function () {
 		clientRender
 	);
 
+	if ( isEnabled( 'marketplace-siteless-checkout' ) ) {
```

Comment:
Let's remove the flag usage. Since we are adding functionality that's not destructive.